### PR TITLE
Fixes legend locking for chorpleth/pointmap with 'auto' color domains

### PR DESCRIPTION
**Pull request by vastcharade**
# Merge Checklist
## :wrench: Issue(s) fixed:
- [ ] Author referenced issue(s) fixed by this PR:
- [ ] Fixes #0

## :smoking: Smoke Test
- [ ] Works in chrome
- [ ] Works in firefox
- [ ] Works in safari
- [ ] Works in ie edge
- [ ] Works in ie 11

## :ship: Merge
- [ ] author crafted PR's title into release-worthy commit message.

### CODE REVIEW

File: src/chart-addons/stacked-legend.js [hunk 1]
```diff
@@ -63,6 +63,21 @@ function setColorScaleDomain(domain) {
   }
 }
 
+function setColorScaleDomain_v1(domain) {
```
Comment (uyanga-gb) [inline src/chart-addons/stacked-legend.js]: I think we should make the object structure same at some point later for raster charts in immerse and the charting. Needs a minor refactoring.

File: src/chart-addons/stacked-legend.js [hunk 2]
```diff
@@ -135,23 +150,36 @@ export function handleLegendLock({ locked, index = 0 }) {
...
     if (color.legend.locked) {
-      layer.setState(setColorScaleDomain(layer.colorDomain()))
+      layer.setState(setColorScaleDomain_v2(layer.colorDomain()))
     } else {
-      layer.setState(setColorScaleDomain("auto"))
+      layer.setState(setColorScaleDomain_v2("auto"))
+      redraw = true
+    }
+  } else if (typeof color.scale === "undefined" && typeof color.domain !== "undefined") {
+    if (color.legend.locked && color.domain === "auto") {
```
Comment (uyanga-gb) [inline src/chart-addons/stacked-legend.js]: There is another property legend.open besides legend.locked, and I am not sure the difference between those two. Does legend.open used somewhere or is it redundant property from immerse? I see legend.open is assigned for Pointmap and Heat map in immerse.

File: src/chart-addons/stacked-legend.js (hunk 1 above)
Comment (vastcharade) [reply src/chart-addons/stacked-legend.js]: absolutely. Not sure why that wasn't done. It's confusing to use two different schemas. Apparently it diverged with heatmap but wasn't brought in line with the other raster layer types. This requires a refactor. See my last comment in this ticket: https://github.com/mapd/mapd-immerse/issues/3703

File: src/chart-addons/stacked-legend.js (hunk 2 above)
Comment (vastcharade) [reply src/chart-addons/stacked-legend.js]: I do not know. Not familiar with the legend code. It seems like it's redundant.

**Pull request merged**